src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -32,120 +34,61 @@
-    }
-
-    /**
-     * Static factory method, which creates @code{StrimziKafkaContainer} with additional configuration of Kafka broker
-     * @param brokerId broker id
-     * @param additionalKafkaConfiguration additional configuration for Kafka broker
-     * @return instance of @code{StrimziKafkaContainer}
-     */
-    public static StrimziKafkaContainer createWithAdditionalConfiguration(final int brokerId, final Map<String, String> additionalKafkaConfiguration) {
-        return createWithAdditionalConfiguration(brokerId, false, additionalKafkaConfiguration);
+    private int brokerId;
+    private String kafkaVersion;
+    private String strimziTestContainerVersion;
+    private boolean useKraft;
+    private String storageUUID;
+
+    static {
+        LOGICAL_KAFKA_VERSION_ENTITY = new KafkaVersionService();
     }
 
-    /**
-     * Static factory method, which creates @code{StrimziKafkaContainer} with empty additional configuration of Kafka broker
-     * @param brokerId broker id
-     * @return instance of StrimziKafkaContainer
-     */
-    public static StrimziKafkaContainer create(final int brokerId) {
-        return new StrimziKafkaContainer(brokerId, false, Collections.emptyMap());
-    }
-
-    /**
-     * Static factory method, which creates @code{StrimziKafkaContainer} with empty additional configuration of Kafka broker, and use Kraft.
-     * @param brokerId broker id
-     * @return instance of StrimziKafkaContainer
-     */
-    public static StrimziKafkaContainer createWithKraft(final int brokerId) {
-        return new StrimziKafkaContainer(brokerId, true, Collections.emptyMap());
+    private void buildDefaults() {
+        String strimziTestContainerVersion;
+        if (this.strimziTestContainerVersion == null || this.strimziTestContainerVersion.isEmpty()) {
+            strimziTestContainerVersion = LOGICAL_KAFKA_VERSION_ENTITY.latestRelease().getStrimziTestContainerVersion();
+            LOGGER.info("You did not specify Strimzi test container version. Using latest release:{}", strimziTestContainerVersion);
+        } else {
+            strimziTestContainerVersion = this.strimziTestContainerVersion;
+        }
+        String kafkaVersion;
+        if (this.kafkaVersion == null || this.kafkaVersion.isEmpty()) {
+            kafkaVersion = LOGICAL_KAFKA_VERSION_ENTITY.latestRelease().getVersion();
+            LOGGER.info("You did not specify Kafka version. Using latest release:{}", kafkaVersion);
+        } else {
+            kafkaVersion = this.kafkaVersion;
+        }
+        this.storageUUID = this.storageUUID == null ? Utils.randomUuid().toString() : this.storageUUID;
+
+        // we need this shared network in case we deploy StrimziKafkaCluster which consist of `StrimziKafkaContainer`
+        // instances and by default each container has its own network, which results in `Unable to resolve address: zookeeper:2181`
+        super.setNetwork(Network.SHARED);
```

Comment:
Any reason for the explicit `super`?

Reply:
I think it does not matter whether I use explicit `super` or `this`. Because I did not override that method I am just using it for clarity that this method is from the `parent`.